NetworkPkg/WifiConnectionManagerDxe/WifiConnectionMgrImpl.c

```diff
@@ -1491,6 +1491,18 @@ WifiMgrOnTimerTick (
   Nic = (WIFI_MGR_DEVICE_DATA *)Context;
   NET_CHECK_SIGNATURE (Nic, WIFI_MGR_DEVICE_DATA_SIGNATURE);
 
+  if ((Nic->ConnectPendingNetwork == NULL) || (StrLen (Nic->ConnectPendingNetwork->SSId) < SSID_MIN_LEN)) {
+    DEBUG ((DEBUG_INFO, "[WiFi Connection Manager] No valid profile for connection, no scan triggered!\n"));
+    gBS->CloseEvent (Event);

```

Comment:
Why do these error return paths close the event and others in this function do not?

Reply:
These failure conditions indicate that the scan is invalid and should not be executed, so we close the event. Others are possible valid scans where there was a hiccup in the system where the scan is still valid but didn't execute fully this round, but should still scan.

Reply:
Perhaps some comments to explain why the event is being closed.

Reply:
`WifiMgrOnTimerTick()` is called by a timer event, and IIUC, closing it will prevent it from being called again until the driver unbinds and binds again. Is that really the right thing to do here?